topsites.py

```diff
@@ -0,0 +1,180 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import base64
+import collections
+import datetime
+import hashlib
+import hmac
+import os
+import requests
+import sys
+import time
+
+from sqlalchemy import create_engine
+from sqlalchemy import Column
+from sqlalchemy import Integer
+from sqlalchemy import String
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import sessionmaker
+from urllib import urlencode
+from xml.dom.minidom import parseString
+
+from webcompat import app
+
+# Constant for alexa top site API
+ATS_ACTION_NAME = "TopSites"
+ATS_RESPONSE_GROUP_NAME = "Country"
+ATS_SERVICE_HOST = "ats.amazonaws.com"
+ATS_AWS_BASE_URL = "http://" + ATS_SERVICE_HOST + "/?"
+
+ATS_DATEFORMAT = "%Y-%m-%dT%H:%M:%S.{}Z"
+ATS_HASH_ALGORITHM = "HmacSHA256"
+
+# Regions/Countries to dump to topsites.db
+REGIONS = {
+    "": (10000, 1),  # Global
+    "US": (1000, 2),
+    "FR": (1000, 2),
+    "IN": (1000, 2),
+    "DE": (1000, 2),
+    "TW": (1000, 2),
+    "ID": (1000, 2),
+    "HK": (1000, 2),
+    "SG": (1000, 2),
+    "PL": (1000, 2),
+    "GB": (1000, 2),
+    "RU": (1000, 2),
+}
+
+# Cache parsed sites, change priority if raised
+topsites = {}
+
+engine = create_engine("sqlite:///" + os.path.join(
+    app.config["BASE_DIR"], "topsites.db"))
+Base = declarative_base()
+Session = sessionmaker(bind=engine)
+session = Session()
+
+
+class Site(Base):
+    '''Define a sqlalchemy base object for a alexa top site.'''
```

Comment:
I think SQLAchemy base object is necessary. Even if it looks like do nothing, but SQLAchemy helps to update and commit value to DB. I just need to update the variables.
For the TopSites class, yes, maybe it could be move to functions instead.